src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/bfaction/BfActionReport.java

```diff
@@ -0,0 +1,115 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.reports.bfaction;
+
+import com.google.common.base.Strings;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import uk.gov.hmcts.ecm.common.helpers.UtilHelper;
+import uk.gov.hmcts.ecm.common.model.ccd.CaseData;
+import uk.gov.hmcts.ecm.common.model.ccd.SubmitEvent;
+import uk.gov.hmcts.ecm.common.model.ccd.items.BFActionTypeItem;
+import uk.gov.hmcts.ecm.common.model.listing.ListingData;
+import uk.gov.hmcts.ecm.common.model.listing.ListingDetails;
+import uk.gov.hmcts.ecm.common.model.listing.items.BFDateTypeItem;
+import uk.gov.hmcts.ecm.common.model.listing.items.BFDateTypeItemComparator;
+import uk.gov.hmcts.ecm.common.model.listing.types.BFDateType;
+import uk.gov.hmcts.ethos.replacement.docmosis.helpers.ReportHelper;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.BROUGHT_FORWARD_REPORT;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.OLD_DATE_TIME_PATTERN;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.OLD_DATE_TIME_PATTERN2;
+
+@Service
```

Comment:
If you use @Service then you are making the class a Spring bean. In this case you inject an instance of the class as a dependency. However you are using the new operator to create an instance of the class in ListingService. I suspect you know all this and the @Service annotation is a mistake. However, this is a pretty fundamental part of using the Spring framework

Reply:
Having a look

Reply:
Corrected. Also, point taken.